storage/watchable_store_bench_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"testing"
+)
+
+func BenchmarkWatchableStoreUnsynced(b *testing.B) {
+	s := newWatchableStore(tmpPath)
+	defer cleanup(s, tmpPath)
+
+	for i := 0; i < 300; i++ {
+		s.Put([]byte(fmt.Sprint("foo", i)), []byte("bar"))
+	}
+
+	b.ReportAllocs()
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		_, cancel := s.Watcher([]byte(fmt.Sprint("foo", i)), true, -1, 0)
+		for i := 0; i < 300; i++ {
+			_, cancel = s.Watcher([]byte(fmt.Sprint("foo", i)), true, -1, 0)
+		}
+		cancel()
```

Comment:
Sorry, I should have been clearer about what I am trying to benchmark.
I wanted to benchmark performance of method `s.Watcher` that appends 
to `slice` or updates the map, and `Cancel` for same operations.

I was not sure how it would get called in real-life example, assuming that
`cancel` gets called only once in awhile. Maybe I am wrong. Is `cancel` something
like the `cancel` in `context` that gets called with `defer` every time? I will come back
to this with running cancel for every case, tonight.

Thanks for feedback.